From e00ed3c8e463bf3a6da549ad6e3a69197a48366f Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 02 Dec 2020 11:14:10 +0200
Subject: [PATCH] [SW-1833] sha256: remove input len limit

sha256 calc wrapper function is limited to minimum input of 32. It seems
there is no such limit in the algorithm itself so the limit can be
discarted.

Change-Id: If3635809624f6e1421889e9af8d47ed0b141afee
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/common/sha256.c b/common/sha256.c
index 12682d9..b696d68 100644
--- a/common/sha256.c
+++ b/common/sha256.c
@@ -261,10 +261,6 @@
 {
 	struct sha256_context ctx;
 
-	if (ilen < SHA256_SUM_LEN) {
-		*output = 0;
-		return;
-	}
 	sha256_starts(&ctx);
 	sha256_update(&ctx, input, ilen);
 	sha256_finish(&ctx, output);
